omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java

```diff
@@ -124,11 +128,14 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
 				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+				if (user == null) {
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.question.fill");
+					request.setAttribute("secretQuestion", getRandomFakeSecretQuestion(username));
```

Comment:
Why ask them a fake question? I would just keep prompting them for the username? Is this the general behavior on other sites?